app/views/non_hmis_clients/_shelter_location_form.haml

```diff
@@ -0,0 +1,29 @@
+- if @non_hmis_client.persisted?
+  - url = send("shelter_location_#{client_type}_client_path", @non_hmis_client.id)
```

Comment:
I'd rather the url come from a helper method than dynamic programming, but not enough to argue about it.